node/src/accountant/db_access_objects/sent_payable_dao.rs

```diff
@@ -280,67 +279,96 @@ impl SentPayableDao for SentPayableDaoReal<'_> {
             return Err(SentPayableDaoError::EmptyInput);
         }
 
-        let sql = "UPDATE sent_payable SET
-               tx_hash = ?1,
-               receiver_address = ?2,
-               amount_high_b = ?3,
-               amount_low_b = ?4,
-               timestamp = ?5,
-               gas_price_wei_high_b = ?6,
-               gas_price_wei_low_b = ?7,
-               nonce = ?8,
-               block_hash = ?9,
-               block_number = ?10
-               WHERE nonce = ?11";
+        let build_case = |value_fn: fn(&Tx) -> String| {
+            new_txs
+                .iter()
+                .map(|tx| format!("WHEN nonce = {} THEN {}", tx.nonce, value_fn(tx)))
+                .collect::<Vec<_>>()
```

Comment:
The collect() is redundant, the join() fn can replace its function entirely

Reply:
The `collect()` is necessary.